internal/client/db.go

```diff
@@ -486,56 +489,55 @@ func (db *DB) prepareToSend(ba *roachpb.BatchRequest) *roachpb.Error {
 
 // send runs the specified calls synchronously in a single batch and returns
 // any errors. Returns (nil, nil) for an empty batch.
-func (db *DB) send(ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.Error) {
+func (db *DB) send(
+	ctx context.Context, ba roachpb.BatchRequest,
+) (*roachpb.BatchResponse, *roachpb.Error) {
 	if len(ba.Requests) == 0 {
 		return nil, nil
 	}
 	if pErr := db.prepareToSend(&ba); pErr != nil {
 		return nil, pErr
 	}
 
-	br, pErr := db.sender.Send(context.TODO(), ba)
+	br, pErr := db.sender.Send(ctx, ba)
 	if pErr != nil {
 		if log.V(1) {
-			log.Infof(context.TODO(), "failed batch: %s", pErr)
+			log.Infof(ctx, "failed batch: %s", pErr)
 		}
 		return nil, pErr
 	}
 	return br, nil
 }
 
-// Runner only exports the Run method on a batch of operations.
-type Runner interface {
-	// Run executes the operations queued up within a batch. Before executing any
-	// of the operations the batch is first checked to see if there were any errors
-	// during its construction (e.g. failure to marshal a proto message).
-	//
-	// The operations within a batch are run in parallel and the order is
-	// non-deterministic. It is an unspecified behavior to modify and retrieve the
-	// same key within a batch.
-	//
-	// Upon completion, Batch.Results will contain the results for each
-	// operation. The order of the results matches the order the operations were
-	// added to the batch.
-	Run(b *Batch) error
-}
-
-func runOneResult(r Runner, b *Batch) (Result, error) {
-	if err := r.Run(b); err != nil {
+// getOneErr returns the error for a single-request Batch that was run.
+// runErr is the error returned by Run, b is the Batch that was passed to Run.
+func getOneErr(runErr error, b *Batch) error {
+	if runErr != nil && len(b.Results) > 0 {
+		return b.Results[0].Err
```

Comment:
This is what `runOneResult` (now `getOneReesult`) does..  I just added another function that's more convenient (in the majority of cases we just need the error). I am guessing the error for the request is more precise than the batch error.